src/vcpkg/portfileprovider.cpp

```diff
@@ -211,158 +490,58 @@ namespace vcpkg
 
         struct OverlayProviderImpl : IFullOverlayProvider
         {
-            OverlayProviderImpl(const ReadOnlyFilesystem& fs, View<Path> overlay_ports)
-                : m_fs(fs), m_overlay_ports(overlay_ports.begin(), overlay_ports.end())
+            OverlayProviderImpl(const ReadOnlyFilesystem& fs, const OverlayPortPaths& overlay_port_paths)
+                : m_fs(fs), m_overlay_index(overlay_port_paths)
             {
-                for (auto&& overlay : m_overlay_ports)
+                for (auto&& overlay : overlay_port_paths.overlay_port_dirs)
                 {
-                    Debug::println("Using overlay: ", overlay);
+                    Debug::println("Using overlay-dir: ", overlay);
 
                     Checks::msg_check_exit(VCPKG_LINE_INFO,
                                            vcpkg::is_directory(m_fs.status(overlay, VCPKG_LINE_INFO)),
```

Comment:
I think we should get rid of this `is_directory` check but that would be product change I didn't need to make so I left it.